File: Arrays/1491. Average Salary Excluding the Minimum and Maximum Salary/1491. Average Salary Excluding the Minimum and Maximum Salary.js
/**
 * @param {number[]} salary
 * @return {number}
 */
var average = function (salary) {
  const min = Math.min(...salary);
  const max = Math.max(...salary);
  const sum = salary.reduce((a, b) => a + b) - min - max;
  return sum / (salary.length - 2);
};

// Time: O(n)
// Space: O(1)

File: Arrays/1491. Average Salary Excluding the Minimum and Maximum Salary/1491. Average Salary Excluding the Minimum and Maximum Salary.py
class Solution:
    def average(self, salary: List[int]) -> float:
        minn = min(salary)
        maxx = max(salary)
        summ = sum(salary) - minn - maxx
        return summ / (len(salary) - 2)

# Time: O(n)
# Space: O(1)